kroxylicious-test-tools/src/main/java/io/kroxylicious/test/client/KafkaClientHandler.java

```diff
@@ -18,26 +20,57 @@
  * Sends a single request frame, waits for a response then closes the channel
  */
 public class KafkaClientHandler extends ChannelInboundHandlerAdapter {
-    private final DecodedRequestFrame<?> decodedRequestFrame;
-    private final CompletableFuture<DecodedResponseFrame<?>> onResponse = new CompletableFuture<>();
+    private final List<DecodedRequestFrame<?>> decodedRequestFrames;
+    private final DecodedRequestFrame<?> initialRequest;
+    List<DecodedResponseFrame<?>> responses = new ArrayList<>();
+    private final CompletableFuture<List<DecodedResponseFrame<?>>> onResponse = new CompletableFuture<>();
+    private final CompletableFuture<Void> initialResponseReceived = new CompletableFuture<>();
 
     /**
      * Creates a KafkaClientHandler
-     * @param decodedRequestFrame the single request to send
+     * @param decodedRequestFrames the requests to send
      */
-    public KafkaClientHandler(DecodedRequestFrame<?> decodedRequestFrame) {
-        this.decodedRequestFrame = decodedRequestFrame;
+    public KafkaClientHandler(DecodedRequestFrame<?> initialRequest, List<DecodedRequestFrame<?>> decodedRequestFrames) {
```

Comment:
Ctor API looks slight odd. Maybe just a list of DRFs with a flag to indicate whether to treat the first as special would be better.